src/components/Books.js

```diff
@@ -1,18 +1,21 @@
 import React, { useState } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { useDispatch } from 'react-redux';
+import { addBook } from '../redux/books/booksSlice';
 
-import PropTypes from 'prop-types';
-
-const Books = ({ addBook }) => {
+const Books = () => {
+  const dispatch = useDispatch();
   const [title, setTitle] = useState('');
   const [author, setAuthor] = useState('');
 
   const handleSubmit = (e) => {
     e.preventDefault();
-    addBook({
-      id: Date.now(),
+    const book = {
+      id: uuidv4(),
```

Comment:
[Optional] Considering that this is a component holding the form for creating books and not the book component itself, I think it'll be more relatable to use a name such as `BookForm` instead of `Book` as the name of the component.